### ISSUE TITLE
precision argument of BackendEstimatorV2.run should account for observable norm

### ISSUE THREAD

**Issue by kevinsung** [Qiskit/qiskit#12414]
### What should we add?

The [current implementation](https://github.com/Qiskit/qiskit/blob/c6c45a1de9140d3051f34a90a8486e689f0a8749/qiskit/primitives/backend_estimator_v2.py#L165) sets `shots = 1 / precision**2`. This is sufficient for a single Pauli operator with norm 1, but it is insufficient for a Hamiltonian with more terms and greater norm. I think it should be `A / precision**2` where `A` is the sum of the absolute value squared of the coefficients of the Hamiltonian, `sum(np.abs(sparse_pauli_op.coeffs)**2)`.

**Comment by ihincks** [comment]
I think that this issue should be filed against particular EstimatorV2 implementations, rather than the base class. This is because the base class can't really enforce how any child chooses to deal with precision. Note that "precision" should always be interpreted as "give me your best guess attempt to satisfy this precision".

For example, the StatevectorSimulator has no notion of shots whatsoever, because it just grabs observable estimates straight from the state vector, and sets the stds to 0 because they're perfect estimates.

The other thing to point out is that `shots = A / precision**2` is not always the right thing to do even if the simulation is shot based, because effects like drift might make it so that shot noise is not the only source of noise affecting precision.

**Comment by kevinsung** [comment]
I agree that `shots = A / precision**2` where A scales with the Hamiltonian norm is not perfect, but I think it's a better representation of a "best guess attempt" than setting A = 1, so I think it makes sense to update the default implementation to use it.

**Comment by ihincks** [comment]
But what's the default implementation? If `qiskit_ibm_runtime.EstimatorV2`, then we can move this issue over there, or to a server-side implementation. If `BackendEstimatorV2` or something in Aer, we can also move or specify.

**Comment by kevinsung** [comment]
Oh I see what you mean. I've changed the title of this issue because it actually refers to BackendEstimatorV2 specifically.

**Comment by kevinsung** [comment]
I opened https://github.com/Qiskit/qiskit-ibm-runtime/issues/1684 for the implementation in Runtime. It appears that Runtime uses their own implementation of BackendEstimatorV2.
